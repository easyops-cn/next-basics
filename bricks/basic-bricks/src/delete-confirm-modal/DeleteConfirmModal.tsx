import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { Modal } from "antd";

import { NS_BASIC_BRICKS, K } from "../i18n/constants";

interface DeleteConfirmModalProps {
  name: string;
  visible: boolean;
  handleConfirm: () => void;
  handleCancel: () => void;
  title: string;
  message?: string;
}

export function DeleteConfirmModal(
  props: DeleteConfirmModalProps
): React.ReactElement {
  const { t } = useTranslation(NS_BASIC_BRICKS);

  const contentTemplate = () => {
    return props.message ? (
      <div>{props.message}</div>
    ) : (
      <div>
        确定删除<span style={{ color: "#FC5043" }}> {props.name} </span>?
      </div>
    );
  };

  useEffect(() => {
    if (props.visible) {
      Modal.confirm({
        title: props.title,
        content: contentTemplate(),
        okText: "确定",
        okType: "danger",
        onOk: props.handleConfirm,
        cancelText: "取消",
        onCancel: props.handleCancel
      });
    }
  }, [props.visible]);

  return <div></div>;
}
