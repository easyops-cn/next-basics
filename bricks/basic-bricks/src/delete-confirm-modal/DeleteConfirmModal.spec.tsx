import React from "react";
import { mount } from "enzyme";
import { DeleteConfirmModal } from "./DeleteConfirmModal";

describe("DeleteConfirmModal", () => {
  it("should work", () => {
    const wrapper = mount(
      <DeleteConfirmModal
        visible={false}
        name={"hello"}
        handleCancel={jest.fn}
        handleConfirm={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ visible: true });
    wrapper.update();
  });
});
