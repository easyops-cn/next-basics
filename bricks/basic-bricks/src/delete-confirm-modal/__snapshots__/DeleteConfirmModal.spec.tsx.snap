// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmModal should work 1`] = `
<DeleteConfirmModal
  handleCancel={[Function]}
  handleConfirm={[Function]}
  name="hello"
  visible={false}
>
  <div />
</DeleteConfirmModal>
`;
